src/services/validateZendeskRequest.spec.ts

```diff
@@ -0,0 +1,412 @@
+    return buildValidRequestBodyWithIds(
+      'session_id',
+      'sessionId1 sessionId2 sessionId3',
+      '',
+      dataPaths
+    )
+  }
+
+  const buildRequestBodyWithDates = (dateFrom: string, dateTo: string) => {
+    const requestBody = buildValidRequestBodyWithIds('session_id', 'sessionId1')
+    requestBody.dateFrom = dateFrom
+    requestBody.dateTo = dateTo
+    return requestBody
+  }
+
+  const buildValidRequestBody = () =>
+    buildValidRequestBodyWithIds('session_id', 'sessionId1')
+
+  const runValidationWithInvalidRequestBody = (requestBody: string | null) => {
+    const validationResult = validateZendeskRequest(requestBody)
+    expect(validationResult.isValid).toEqual(false)
+    expect(validationResult.validationMessage).toEqual('No data in request')
+  }
+
+  const getTodayUtc = (): number => {
+    const today = new Date()
+    return Date.UTC(today.getFullYear(), today.getMonth(), today.getDate())
+  }
+
+  const getTodayPlusDaysAsString = (days: number): string => {
+    const date = new Date(getTodayUtc())
+    date.setDate(date.getDate() + days)
+    return getDateAsString(date)
+  }
+
+  const getTodayAsString = () => getDateAsString(new Date(getTodayUtc()))
+
+  const getDateAsString = (date: Date): string =>
+    `${date.getFullYear()}-${(date.getMonth() + 1)
+      .toString()
+      .padStart(2, '0')}-${date.getDate().toString().padStart(2, '0')}`
+
+  it('should return an invalid response if request body is null', () => {
+    runValidationWithInvalidRequestBody(null)
+  })
+
+  it('should return an invalid response if request body is blank', () => {
+    runValidationWithInvalidRequestBody('')
+  })
+
+  it('should return an invalid response if request body is malformed', () => {
+    runValidationWithInvalidRequestBody('hello')
+  })
+
+  it('should parse data into response if request data is valid', () => {
+    console.log(
+      'here is the request',
+      JSON.stringify(
+        buildValidRequestBodyWithIds(
+          'session_id',
+          'sessionId1 sessionId2 sessionId3',
+          'dob name passport_number'
+        )
+      )
+    )
+
+    const validationResult = validateZendeskRequest(
+      JSON.stringify(
+        buildValidRequestBodyWithIds(
+          'session_id',
+          'sessionId1 sessionId2 sessionId3',
+          'dob name passport_number'
+        )
+      )
+    )
+    expect(validationResult.isValid).toEqual(true)
+
+    expect(validationResult.dataRequestParams?.dateFrom).toEqual('2021-08-01')
+    expect(validationResult.dataRequestParams?.dateTo).toEqual('2021-08-01')
+    expect(validationResult.dataRequestParams?.zendeskId).toEqual(testZendeskId)
+    expect(validationResult.dataRequestParams?.resultsEmail).toEqual(
+      testValidResultsEmail
+    )
+    expect(validationResult.dataRequestParams?.resultsName).toEqual(
+      testResultsName
+    )
+    expect(validationResult.dataRequestParams?.identifierType).toEqual(
+      'session_id'
+    )
```

Comment:
I see the next TC checks for specific values, but I'd suggest we at least check the list is not empty at this point.